src/osemgrep/cli_scan/Scan_CLI.ml

```diff
@@ -876,25 +872,39 @@ let cmdline_term ~allow_empty_config : conf Term.t =
       | _else_ -> default.output_format
     in
     let engine_type =
-      match (oss, pro_lang, pro_intrafile, pro) with
-      | false, false, false, false when secrets ->
-          Engine_type.(PRO Language_only)
-      | false, false, false, false -> default.engine_type
+      (* This first bit just rules out mutually exclusive options. *)
+      (match (oss, pro_lang, pro_intrafile, pro) with
       | true, false, false, false when secrets ->
           Error.abort
             "Mutually exclusive options --oss/--beta-testing-secrets-enabled"
-      | true, false, false, false -> OSS
-      | false, true, false, false -> PRO Engine_type.Language_only
-      | false, false, true, false -> PRO Engine_type.Intrafile
-      | false, false, false, true -> PRO Engine_type.Interfile
+      | false, false, false, false
```

Comment:
you can also count the number of true in the list. 
let cnt = [oss; pro_lang; pro_intrafile; pro] |> Common.filter (fun b -> b) |> List.length in